### PR TITLE
Add ibmdb2_conf and ibmdb2_session resource

### LINKED ISSUE #5591: Add resources for IBM DB2 support
## Detailed Description
(i)Use db2_session Chef InSpec audit resource to test json queries run against a IBM DB2.

(ii)Use db2_ conf Chef InSpec audit resource to test the contents of the configuration file for IBM DB2

## Context
Needed for CIS benchmark and customer needs.

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

This PR adds basic support for IBM Db2 by adding ibmdb2_conf and ibmdb2_session resources. 
- Supports: Unix, Windows
- Includes documentation.
- Manual testing done with ibmdb2 docker container, Ubuntu 20, Windows Server 2019
## Related Issue

Fix #5591 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/resources/ibmdb2_conf.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require "inspec/resources/ibmdb2_conf"
```
Comment (Nik08) [inline lib/inspec/resources/ibmdb2_conf.rb]: Seems like this is the self required in this resource? Do we need this?

File: lib/inspec/resources/ibmdb2_session.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require "inspec/resources/ibmdb2_session"
```
Comment (Nik08) [inline lib/inspec/resources/ibmdb2_session.rb]: Here too, do we need this too?

File: lib/inspec/resources/ibmdb2_conf.rb (hunk 1 above)
Comment (Vasu1105) [reply lib/inspec/resources/ibmdb2_conf.rb]: Thanks. That should not be there.

**Comment by Nik08** [comment]
LGTM @Vasu1105.

**Comment by dkumaras** [comment]
Pls check and let us know if the edits are okay.

**Comment by Vasu1105** [comment]
> Pls check and let us know if the edits are okay.
@dkumaras 
Some indentation changes are breaking formatting. This is before changes https://github.com/inspec/inspec/blob/084f6b1f2b9f4b56716cbc4dc0a5c07db07dc4b5/docs-chef-io/content/inspec/resources/ibmdb2_conf.md

and this is after changes.
https://github.com/inspec/inspec/blob/1794716c0461df963afb979c8c586e25838f76a4/docs-chef-io/content/inspec/resources/ibmdb2_conf.md

File: docs-chef-io/content/inspec/resources/ibmdb2_conf.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  its("output") { should include("Audit buffer size (4KB) (AUDIT_BUF_SZ) = 0")}
+end
+
+Windows
+
+describe ibmdb2_conf do
+  its("output") { should_not be_empty }
+  its("output") { should include("Audit buffer size (4KB) (AUDIT_BUF_SZ) = 0")}
+end
+
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/ibmdb2_conf.md:37]: Is there a way to do this with `db_name`, or is this required with `db2_executable_file_path` and `db_instance`?

For example:

```ruby
describe ibmdb2_conf(db_name: "NAME") do
    test things
end
```

or

```ruby
describe ibmdb2_conf(db2_executable_file_path: "/opt/ibm/db2/V11.5/bin/db2", db_instance: "db2inst1", db_name: "name") do
  test things
end
```

Comment (Vasu1105) [reply docs-chef-io/content/inspec/resources/ibmdb2_conf.md:37]: Its not required in case of Windows.

Comment (IanMadd) [reply docs-chef-io/content/inspec/resources/ibmdb2_conf.md:37]: Is `db_name` required for linux?

Comment (Vasu1105) [reply docs-chef-io/content/inspec/resources/ibmdb2_conf.md:37]: for ibmdb2_conf resource, db_name is not required for both Linux and windows.

**Pull request merged**